src/main/java/mate.academy/Main.java

```diff
@@ -4,13 +4,16 @@
 import mate.academy.lib.Injector;
 import mate.academy.model.Product;
 import mate.academy.service.ProductService;
+import mate.academy.service.impl.ProductServiceImpl;
 
 public class Main {
 
     public static void main(String[] args) {
         // Please test your Injector here. Feel free to push this class as a part of your solution
         Injector injector = Injector.getInjector();
-        ProductService productService = null;
+        ProductService productService = (ProductService) injector
+                .getInstance(ProductServiceImpl.class);
+
         List<Product> products = productService.getAllFromFile("products.txt");
```

Comment:
Directly using the filename `"products.txt"` in the `getAllFromFile` method call makes the code less flexible. Consider defining the filename as a constant or allowing it to be passed as an argument to your application to improve flexibility and reusability.